samples/bluetooth/fast_pair/locator_tag/include/app_fp_adv.h

```diff
@@ -32,11 +32,51 @@ enum app_fp_adv_mode {
 	APP_FP_ADV_MODE_NOT_DISCOVERABLE,
 };
 
-/** Set the Fast Pair advertising mode.
+/** Fast Pair advertising trigger. */
+struct app_fp_adv_trigger {
+	const char *id;
+	bool enabled;
+};
+
+/** Register a trigger for Fast Pair advertising.
+ *
+ *  @param _name Trigger structure name.
+ *  @param _id Trigger identifier.
+ */
+#define APP_FP_ADV_TRIGGER_REGISTER(_name, _id)				\
+	BUILD_ASSERT(_id != NULL);					\
+	static STRUCT_SECTION_ITERABLE(app_fp_adv_trigger, _name) = {	\
+		.id = _id,						\
+	}
+
+/** Request turning on or off the Fast Pair advertising.
+ *
+ *  The Fast Pair advertising will be turned off only when no
+ *  trigger has requested to keep the Fast Pair advertising on.
+ *
+ *  If the current Fast Pair advertising mode is the same as the requested mode,
+ *  the Fast Pair advertising will not be refreshed.
+ *  Use the @ref app_fp_adv_refresh function to manually refresh the Fast Pair
+ *  advertising.
+ *
+ *  @param trigger Trigger identifier.
+ *  @param enable true to request to turn on the Fast Pair advertising.
+ *		  false to remove request to turn on the Fast Pair advertising.
+ */
+void app_fp_adv_request(struct app_fp_adv_trigger *trigger, bool enable);
+
+/** Refresh the Fast Pair advertising.
+ *
+ *  This function is used to manually trigger the refreshing of the Fast Pair
+ *  advertising data to ensure that the advertising data is up to date.
```

Comment:
Write a cautionary note about using this API with care as updating the advertising payload asynchronously to the RPA address and the FMDN extension.